natter-api/src/main/java/com/manning/apisecurityinaction/Main.java

```diff
@@ -14,6 +14,7 @@
 public class Main {
```

Comment:
Main code review:

1. Authentication is missing.
2. Passwords must always be hashed to protect the system.
3. Set Username and setpassword methods could be used to set username and password 
For example:
(JdbcConnectionSource)connectionSource).setUsername("natter");
((JdbcConnectionSource)connectionSource).setPassword("password");
4. Must check if the datasource exists.
5. No timestamp is added on the header in API requests.(The server will compare the current timestamp to the request timestamp and only accepts the request if it is within a reasonable timeframe (1-2 minutes, perhaps)).